Brokerages/GDAX/GDAXBrokerage.cs

```diff
@@ -290,6 +295,130 @@ public override List<Cash> GetCashBalance()
+        private IEnumerable<TradeBar> GetHistoryFromCandles(HistoryRequest request)
+        {
+            var productId = ConvertSymbol(request.Symbol);
+            var granularity = Convert.ToInt32(request.Resolution.ToTimeSpan().TotalSeconds);
+
+            var startTime = request.StartTimeUtc;
+            var endTime = request.EndTimeUtc;
+            var maximumRange = TimeSpan.FromSeconds(MaxDataPointsPerHistoricalRequest * granularity);
+
+            do
+            {
+                var maximumEndTime = startTime.Add(maximumRange);
+                if (endTime > maximumEndTime)
+                {
+                    endTime = maximumEndTime;
+                }
+
+                var restRequest = new RestRequest($"/products/{productId}/candles?start={startTime:o}&end={endTime:o}&granularity={granularity}", Method.GET);
+                var response = ExecuteRestRequest(restRequest, GdaxEndpointType.Public);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "HistoryError", 
+                        $"History request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}"));
+                    yield break;
+                }
+
+                var bars = ParseCandleData(request.Symbol, granularity, response.Content, startTime);
+
+                TradeBar lastPointReceived = null;
+                foreach (var datapoint in bars.OrderBy(x => x.Time))
+                {
+                    lastPointReceived = datapoint;
+                    yield return datapoint;
+                }
+
+                startTime = lastPointReceived?.EndTime ?? request.EndTimeUtc;
+                endTime = request.EndTimeUtc;
+            } while (startTime < request.EndTimeUtc);
+        }
+
+        /// <summary>
+        /// Parse TradeBars from JSON response
+        /// https://docs.pro.coinbase.com/#get-historic-rates
+        /// </summary>
+        private static IEnumerable<TradeBar> ParseCandleData(Symbol symbol, int granularity, string data, DateTime startTimeUtc)
+        {
+            if (data.Length == 0)
+            {
+                yield break;
+            }
+
+            var parsedData = JsonConvert.DeserializeObject<string[][]>(data);
+            var period = TimeSpan.FromSeconds(granularity);
+
+            foreach (var datapoint in parsedData)
+            {
+                var time = Time.UnixTimeStampToDateTime(double.Parse(datapoint[0], CultureInfo.InvariantCulture));
+
+                if (time < startTimeUtc)
+                {
+                    // Note from GDAX docs:
+                    // If data points are readily available, your response may contain as many as 300 candles
+                    // and some of those candles may precede your declared start value.
+                    yield break;
```

Comment:
`parsedData` is ordered from by time descending?

Reply:
Correct, the `candles` endpoint always returns data ordered by time descending and there is no query parameter to change the ordering.